main.go

```diff
@@ -93,14 +96,27 @@ func main() {
 	}
 
 	// Migrate the schema
+	err = migration.Migrate(db)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	// Nothing to here except exit, since the migration is already performed.
+	if migrateDB {
+		os.Exit(0)
+	}
+
 	ts := models.NewGormTransactionSupport(db)
 	witRepo := models.NewWorkItemTypeRepository(ts)
 	wiRepo := models.NewWorkItemRepository(ts, witRepo)
 
-	if err := transaction.Do(ts, func() error {
-		return migration.Perform(context.Background(), ts.TX(), witRepo)
-	}); err != nil {
-		panic(err.Error())
+	// Make sure the database is populated with the correct types (e.g. system.bug etc.)
+	if configuration.GetPopulateCommonTypes() {
+		if err := transaction.Do(ts, func() error {
+			return migration.Populate(context.Background(), ts.TX(), witRepo)
```

Comment:
@tsmaeder thank you for reviewing! What do you mean with this sentence?

> I think the transaction isolation level needs to be addressed.

> I really don't understand how we are going to handle the next change to a base work item type. 

As far as I understood @aslakknutsen , the base types (as populated with `migration.PopulateCommonTypes()`) will be used only in almighty online. That's why we didn't put them in the migration. Maybe we can create them with a migration SQL script and get rid of the populate function. Then if a customer doesn't want to use those default types, we run a `DELETE` of those WITs on the DB instead of an `INSERT`. Sound like a plan @tsmaeder and @aslakknutsen ?